src/StoryTime.jsx

```diff
@@ -30,11 +34,72 @@ const MegaChatBox = ({
 }) => {
   const [currentMessage, setCurrentMessage] = useState(message);
   const selectedOptionIndex = options ? options.indexOf(option) : -1;
+  const [ opened, setOpened ] = useState( false );
+
+  const boxTopRef = useRef( null );
+  const boxBottomRef = useRef( null );
+  const boxLeftRef = useRef( null );
+  const boxRightRef = useRef( null );
 
   useEffect(() => {
     if (message && currentMessage !== message) {
-      setCurrentMessage(message);
+        setCurrentMessage(message);
+    }
+
+    //
+
+    boxBottomRef.current.innerHTML = '';
+    boxTopRef.current.innerHTML = '';
+    boxLeftRef.current.innerHTML = '';
+    boxRightRef.current.innerHTML = '';
+
+    const type = ( message ? 'Opening' : 'Closing' );
+
+    lottie.loadAnimation({
+        container: boxBottomRef.current,
+        renderer: 'svg',
+        loop: false,
+        autoplay: true,
+        path: `./lottie/modal/${ type }/bottom.json`
+    });
+
+    lottie.loadAnimation({
+        container: boxTopRef.current,
+        renderer: 'svg',
+        loop: false,
+        autoplay: true,
+        path: `./lottie/modal/${ type }/top.json`
+    });
+
+    lottie.loadAnimation({
+        container: boxLeftRef.current,
+        renderer: 'svg',
+        loop: false,
+        autoplay: true,
+        path: `./lottie/modal/${ type }/left.json`
+    });
+
+    lottie.loadAnimation({
+        container: boxRightRef.current,
+        renderer: 'svg',
+        loop: false,
+        autoplay: true,
+        path: `./lottie/modal/${ type }/right.json`
+    });
+
+    if ( openTimeoutId !== null ) clearTimeout( openTimeoutId );

```

Comment:
Can this be put into a returned cleanup function?

Reply:
Fixed.